### PR TITLE
misc(treemap): shade background for unused bytes

### PR DESCRIPTION
**Pull request by paulirish**
proposing this alternative UX for 'Unused Bytes' mode.

https://lighthouse-git-treemap-unusedhue-googlechrome.vercel.app/gh-pages/treemap/?debug  and click 'unused bytes'

compared to 

https://lighthouse-git-master-googlechrome.vercel.app/gh-pages/treemap/?debug

### CODE REVIEW

**Comment by connorjclark** [comment]
I think it's too busy... prefer this:

image

also concerned how user would know which shade is unsued vs used

also concerned that we'd have two different ways to visualizing coverage.

would like to see: only highlighed nodes (the most egregious unused nodes) getting a special treatment; and that treatment being the same thing used in the coverage table column

**Comment by adamraine** [comment]
> also concerned how user would know which shade is unsued vs used

I agree, couldn't tell which shade represents "used" at first glance.

**Comment by paulirish** [comment]
> too busy

yes. it's def busy. 

> i prefer this:

yeah the thing i don't like about that is the arbitrary thresholds. I think there are likely still modules of interest that are slightly below those numbers.  it's hard to hit a binary threshold right so i want to make it a continuous thing.

> I agree, couldn't tell which shade represents "used" at first glance.

yeah totally fair. i'm gonna try a candystripe shading to see if it helps.

**Comment by connorjclark** [comment]
+1 to the new changes Paul demoed in today's meeting.

File: lighthouse-treemap/app/src/main.js [hunk 1]
```diff
@@ -412,6 +391,7 @@ class TreemapViewer {
 
     if (rootChanged || viewChanged) {
       this.updateColors();
+      this.el.id = this.currentViewMode.id;
```
Comment (connorjclark) [inline lighthouse-treemap/app/src/main.js:394]: imo this should be a class, `lh-treemap--${...}`

File: lighthouse-treemap/app/src/main.js [hunk 2]
```diff
@@ -612,6 +592,10 @@ class TreemapViewer {
         this.getHueForD1NodeName(depthOneNode ? depthOneNode.name : node.name);
       const depthOneNodeColor = hue !== undefined ? this.getColorFromHue(hue) : 'white';
 
+      // @ts-ignore: webtreemap will add a dom node property to every node.
+      const dom = /** @type {HTMLElement?} */ (node.dom);
```
Comment (connorjclark) [inline lighthouse-treemap/app/src/main.js:596]: I added `dom` to the types in a previous commit. bad merge?

Comment (connorjclark) [reply lighthouse-treemap/app/src/main.js:596]: use `node.dom` directly.

File: lighthouse-treemap/app/styles/treemap.css [hunk 3]
```diff
@@ -201,6 +208,27 @@ header {
   outline: 2px solid black;
 }
 
+.webtreemap-node::before {
```
Comment (connorjclark) [inline lighthouse-treemap/app/styles/treemap.css:211]: magic? magic. ok.

File: lighthouse-treemap/app/styles/treemap.css [hunk 4]
```diff
@@ -201,6 +208,27 @@ header {
...
+  background: repeating-linear-gradient(
+          -45deg,
+          hsla(0, 0%, 0%, 0),
+          hsla(0, 0%, 0%, 0) 2px,
+          hsla(7, 85%, 56%, 0.3) 2px,
+          hsla(7, 85%, 56%, 0.3) 4px
+  );
+}
+
+.lh-treemap#unused-bytes .webtreemap-node::before {
```
Comment (connorjclark) [inline lighthouse-treemap/app/styles/treemap.css:228]: why not just move all of the css here? why need to toggle `display`?

File: lighthouse-treemap/app/src/main.js [hunk 5]
```diff
@@ -612,6 +592,9 @@ class TreemapViewer {
         this.getHueForD1NodeName(depthOneNode ? depthOneNode.name : node.name);
       const depthOneNodeColor = hue !== undefined ? this.getColorFromHue(hue) : 'white';
 
+      const dom = node.dom;
+      if (!dom) return;
```
Comment (connorjclark) [inline lighthouse-treemap/app/src/main.js:596]: use node.dom directly.

File: lighthouse-treemap/app/src/main.js [hunk 6]
```diff
@@ -623,12 +606,16 @@ class TreemapViewer {
...
-        backgroundColor = depthOneNodeColor;
+        dom.style.backgroundColor = backgroundColor;
+        return;
       }
 
-      if (node.dom) {
-        node.dom.style.backgroundColor = backgroundColor;
+      dom.style.backgroundColor = depthOneNodeColor;
+
+      // Set a bicolor background to communicate unused-bytes
```
Comment (connorjclark) [inline lighthouse-treemap/app/src/main.js:615]:
```suggestion
      // Shade the node to communicate coverage.
```

**Review by connorjclark — APPROVED** [review]

**Pull request merged**